### ISSUE TITLE
Cardinality error is incorrect for models derived from DETR that do not have an explicit background class

### ISSUE THREAD

**Issue by jveitchmichaelis** [huggingface/transformers#42199]
##  Issue

For DETR variants, the cardinality errors that are reported during training are incorrect. This was reported in the DeformableDETR repository, and was acknowledged but not resolved:

https://github.com/fundamentalvision/Deformable-DETR/issues/24

Since all the derived models no longer include an explicit background class, the returned cardinality error is wrong. This affects at least Deformable DETR. I'm not sure about Conditional as it doesn't seem to be reproducible in the fine-tuning example.

While `loss_cardinality` is not strictly a loss, it is a bug to include it. It's confusing to users because it's returned as part of the loss dict. It also breaks a common pattern in logging frameworks of calling `logger.log('total_loss', sum(loss_dict.items())`, because `loss_cardinality` is both huge compared to the real losses, and does not change during training in a meaningful way.

## Context

In the original DETR implementation (broadly copied here), the class embedding outputs +1 logits to explicitly model the background (which is tacked on as index `-1`). Hence the recommendation to instantiate models with `max(class_ids)`. In later works, the loss function uses a sigmoid/focal loss and the output logits have len(num_classes). It's mentioned in passing in the DeformableDETR paper, but not clear to me if it was strictly necessary for the decoder design or faster convergence. I'm not aware of an earlier experiment, but every other paper seems to now use that convention.

The ImageLoss function has this method: https://github.com/huggingface/transformers/blob/8cb5963cc22174954e7dca2c0a3320b7dc2f4edc/src/transformers/loss/loss_for_object_detection.py#L143-L157

Note that the assumption is “no object” is explicitly class index -1. In other version, like Deformable DETR, ImageLoss is sub-classed. But since this index actually refers to a genuine class, what this actually computes is the number of predictions that are classes[-1] versus ground truth. As an obvious example, in the case of a 1-class model, the output is always `num_queries - num_gt`.

## Proposed solution

Remove the entry in the `loss_dict` entirely and don't calculate cardinality error. I'm happy to submit a PR for this.

Alternatively we could compute it based on the score threshold, but that's a post processing step and not in the forward pass. There may be a more useful metric to return here, but since these models always predict `num_queries` objects, the concept of cardinality only makes sense once you filter them.

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [x] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [x] My own task or dataset (give details below)

### Reproduction

Run the forward pass with any of these models, including targets. This is a 1-class model:

<img width="1405" height="321" alt="Image" src="https://github.com/user-attachments/assets/a0da05e3-66e2-4a7d-8193-70c5bacba742" />

### Expected behavior

Either return a true cardinality error or nothing. However as above - it doesn't make sense because the model returns all queries as foreground predictions.

**Comment by jveitchmichaelis** [comment]
Had a quick look at the notebook sample. I think that's a good reproduction, though the absolute magnitude of the error is lower. For a 1-class model the logit shape is `[B, queries, 1]`. The notebook is incorrect where it mentions that there is a +1 in the class embedding, probably copied from a DETR example, you can verify by checking the example output logits in the proceeding cell.

For 1 class:

```
pred_idx = outputs.logits.argmax(-1) # 0 as only foreground
empty_idx = outputs.logits.shape[-1] - 1 # 0 as single class
card_pred = (pred_idx != empty_idx).sum(1) # Always 0
card_err = nn.functional.l1_loss(card_pred.float(), target_lengths.float()) # | pred - len(target) | -> len(target)
```

For an `N` class model, `card_err` is returning how many predictions were assigned class -1 vs the number of targets.

**Comment by Rocketknight1** [comment]
cc @nielsrogge maybe?

**Comment by NielsRogge** [comment]
I'll ping @yonigozlan here

**Comment by NielsRogge** [comment]
@yonigozlan any updates on this?

**Comment by yonigozlan** [comment]
I'll have a look! @NielsRogge any idea why we used cardinality + 1 in the first place? I'm missing a bit of context here and don't want to break BC if this was done intentionally

**Comment by jveitchmichaelis** [comment]
Best I can tell is this was something that was an informative metric for (original) DETR and got left in the codebase by projects which built off it.

It made sense then because there was a background class in DETR. Most (all?) DETR-derived models nowadays (Deformable, Conditional, etc) do not have this explicit background and you rely on confidence scores to filter boxes. 

To be clear this should not be removed for DETRForObjectDetection.

**Comment by NielsRogge** [comment]
@yonigozlan this was taken from the [original implementation](https://github.com/facebookresearch/detr/blob/29901c51d7fe8712168b8d0d64351170bc0f83e0/models/detr.py#L37).
